### ISSUE TITLE
Wrong DLL copied to package directory, overwriting correct DLL

### ISSUE THREAD

**Issue by sandersaares** [Azure/service-fabric-issues#54]
A SF actor project references System.Runtime.dll 4.1.0.0 that is included via a NuGet package reference. When I build this project, I see the correct DLL in the bin folder. However, when I deploy it, in the SF package folder, the DLL is now incorrect!

Monitoring closer what happens on deploy, I see that the correct DLL from the actor project's bin folder is copied to the package folder. Then it is overwritten with c:\Program Files (x86)\Reference Assemblies\Microsoft\Framework.NETFramework\v4.6.2\Facades\System.Runtime.dll! This is not the only affected DLL - it appears to affect all those in the Facades folder, which overwrite anything imported via NuGet.

This is a complete roadblock for me - I cannot develop my service, as it breaks on deploy due to the wrong dependency being present. Is there something obvious I am missing here?

Incidentally, I did not find any obvious way to get diagnostics about the startup failure of the app - the default ETW entries (right click and start streaming trace in Cloud Explorer) showed nothing about this. The only way I could even get an exception message was by attaching a debugger.

**Comment by masnider** [comment]
Can you provide what errors you are seeing?

**Comment by sandersaares** [comment]
1. Create new stateless web API application.
1. Upgrade project to .NET Framework 4.6.2.
1. Install package NETStandard.Library.
1. Press F5 to start debugging.
1. Exception occurs.

> An exception of type 'System.IO.FileLoadException' occurred in WebApi1.exe but was not handled in user code
> 
> Additional information: Could not load file or assembly 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Investigation indicates that the wrong DLL is included in the package.

**Comment by masnider** [comment]
Does this happen to you for any general project? Say you just created a new console app and then added the NETStandard.Library package and hit F5?

**Comment by sandersaares** [comment]
No, this does not happen for console applications as console applications do not create Service Fabric packages. This is specific to SF.

**Comment by ericstj** [comment]
Can someone share a project that's reproing the problem?  It looks to me like an inconsistency between the binding redirects used by the app and the references added by NuGet.  /cc @rrelyea

**Comment by MikkelHegn** [comment]
https://github.com/MikkelHegn/Application23 - This Solution repoes the problem.

**Comment by sandersaares** [comment]
The problem still remains after the October 10 SDK update.

**Comment by ericstj** [comment]
Here’s what’s going on.  I had to repro from the commandline with msbuild application1.sfproj /t:Package /flp:v=diag because the call to the Package target from the project system was not honoring my VS settings for MSBuild verbosity (unrelated bug).

The bad façade was coming from ServiceCodeFiles as
```
C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.2\Facades\System.Runtime.dll
        CopyLocal=false
        MSBuildSourceProjectFile=C:\Users\ericstj\Documents\Visual Studio 2015\Projects\Application1\WebApi1\WebApi1.csproj
        MSBuildSourceTargetName=BuiltProjectOutputGroupDependencies
        OriginalItemSpec=..\WebApi1\WebApi1.csproj
        ResolvedFrom=ImplicitlyExpandDesignTimeFacades
        TargetPath=System.Runtime.dll
        WinMDFile=false
```

This comes from the `CollectServiceCodeFiles` target in Microsoft.VisualStudio.Azure.Fabric.Application.targets.

That target is calling the BuiltProjectOutputGroupDependencies on the service project, but not filtering references that set to the copy-local dependencies:
```xml
    <MSBuild
        Projects="$(_ServiceProject)"
        Targets="
          BuiltProjectOutputGroupDependencies;
          DebugSymbolsProjectOutputGroupDependencies;
          DocumentationProjectOutputGroupDependencies;
          SGenFilesOutputGroupDependencies"
        ContinueOnError="false"
        Properties="$(ReferencedProjectCommonBuildProperties)">
      <Output TaskParameter="TargetOutputs" ItemName="OutputGroupsDependencies" />
    </MSBuild>

    <!-- Figure out the dependencies that need to be copied by checking if the dependency exists
         in the target directory. -->
    <ItemGroup>
      <CopyLocalDependencies Include="@(OutputGroupsDependencies)" Condition="Exists('$(ProjectTargetDir)%(Filename)%(Extension)')" />
    </ItemGroup>
```

The broken piece is that check on the include.  It’s determining if it needs to copy the file based solely on the whether or not a file with the same name exists in the output directory.  In this case, the project had two references: one to the targeting pack, which isn’t copy-local, and another to the nuget package, which is copy local.  A fix here would be to add `AND '%(OutputGroupsDependencies.CopyLocal)' != 'false'`, or potentially just require “CopyLocal” to be set to true

I fixed this up in my local copy and it made the problem go away.

**Comment by MikkelHegn** [comment]
This fix is in the 5.4 update for Service Fabric.
